### PR TITLE
Develop on master

### PR DESCRIPTION
**Pull request by arummler**
Fix some compilation issues on v2.0 when compiling with LCIO/EUTELESCOPE and LCIO alone (issue #173).

### CODE REVIEW

**Comment by airqui** [comment]
Hi,
I just made a quick check compiling it and running it:

- it compiles perfectly not using EUTELESCOPE, just compiling with LCIO and compiling the calice producers (see my cmake output at the end of the message)
- it seems that the loading of the configuration files still has some absolute paths hardcoded, so it always points to: ../conf/(whatheverfilewithabsolutepath.conf).conf or similar
- using the same producers and scripts that I was using with eudaq 1.5, there is a problem connecting with the DataColector ("trying to push Empty Processor to batch") but I guess that this is because I am just not doing thinks properly to eduaq2.

I hope that this info will be useful.

Cheers,
Adrian

[calice@flchcallab4 build]$ cmake -DUSE_LCIO=ON -DBUILD_calice=ON ..
-- The C compiler identification is GNU
-- The CXX compiler identification is GNU
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Checking for C++11 compliance of compiler
-- Checking for C++11 compliance of compiler - available
-- Checking to see if CXX compiler accepts flag -std=c++11
-- Checking to see if CXX compiler accepts flag -std=c++11 - yes
-- Using gcc-specific preprocessor identifiers
CMake Warning at CMakeLists.txt:41 (MESSAGE):
  Your do not seem to have fetched the latest tags in your git repository --
  please consider running 'git fetch upstream'


-- Main EUDAQ library directory is clean (all changes have been commited).
-- Determined EUDAQ version v1.4.5+565~g39c6fa4
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Looking for include files CMAKE_HAVE_PTHREAD_H
-- Looking for include files CMAKE_HAVE_PTHREAD_H - found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Could NOT find PROTOBUF (missing:  PROTOBUF_LIBRARY PROTOBUF_INCLUDE_DIR)
-- Looking for Root...
-- Found ROOT: /home/calice/Programs/root/bin/root (found version 5.34/28)
-- Looking for LCIO...
-- Found LCIO: /home/calice/Programs/ilcinstall/v01-17-06/lcio/v02-05/lib/liblcio.so
-- Looking for CMSPixel dependencies: pxarCore...
-- NOT building converter plugin for CMSPixel.
-- Could not find Qt5, searching for Qt4 instead...
-- Looking for Q_WS_X11
-- Looking for Q_WS_X11 - found
-- Looking for Q_WS_WIN
-- Looking for Q_WS_WIN - not found.
-- Looking for Q_WS_QWS
-- Looking for Q_WS_QWS - not found.
-- Looking for Q_WS_MAC
-- Looking for Q_WS_MAC - not found.
-- Found Qt4: /usr/bin/qmake (found version "4.8.5")
-- Found Qt 4.8.5, will build GUI executables
-- Looking for Root...
-- Found ROOT: /home/calice/Programs/root/bin/root (found version 5.34/28)
-- Found Doxygen: /usr/bin/doxygen
-- Building producer: calice
-- CMAKE_BINARY_DIR: /home/calice/eudaq2-arummler/build
-- PROJECT_SOURCE_DIR: /home/calice/eudaq2-arummler
-- This is CMake 2.8.5 (/home/calice/Programs/ilcinstall/v01-17-06/CMake/2.8.5/bin/cmake) running on Linux (3.10.0-229.4.2.el7.x86_64 sh: lsb_release: command not found)
-- 64-bit compiler detected
-- Using GNU (/usr/bin/g++) version 4.8.3
-- Compiler Flags: -fPIC -O2 -std=c++11
-- With COMPILE_DEFINITIONS = USE_EUDAQ2_VERSION;CPP11=1;EUDAQ_PLATFORM=PF_LINUX;EUDAQ_FUNC=__PRETTY_FUNCTION__ ;_GLIBCXX_USE_NANOSLEEP
-- Found PythonInterp: /usr/bin/python2.7
-- *** Build Type RelWithDebInfo ***
-- *** Compiler Flags:  -fPIC -O2 -std=c++11
-- *** Compiler Flags:  -fPIC -O2
-- Configuring done
-- Generating done

**Comment by arummler** [comment]
Hi,
I appreciate it a lot if someone looks at the code.
1.) very good
2.) You mean at the startup before choosing something for the first time? There is only the default directory ../conf set but we can also remove it as soon as it gets deleted from the repo. If we leave it empty I suppose the file dialogue will just default to the home directory (if only linux and similar on other systems). I will try it and make a commit. A conf file shouldn't be set by default.
3.) What did you exactly start? With which parameters? I tried euRun, the TestDataCollector and the calice producer and it all connects correctly.

Cheers,
Andre

File: main/lib/src/FileWriterLCIOC.cc [hunk 1]
```diff
@@ -33,9 +33,10 @@ namespace eudaq {
     // int m_runnumber; 
   };
 
-  namespace {
-    static RegisterFileWriter<FileWriterLCIOC> reg("calice");
-  }
+  //namespace {
```
Comment (simonspa) [inline main/lib/src/FileWriterLCIOC.cc]: You could probably remove those commented lines directly instead of having dead code laying around.

**Comment by airqui** [comment]
Hi Andre,

Thanks for your answers and for the developments.
I tried again. Here are some feedbacks:

2) Yes, the configuration file that you load when you use the Load button is: ../conf/+AbsolutePathToYourComfFile.conf+.conf Is easily fixable, if after chosing the file with the load option, you rewrite it accordingly. I gues that, for the moment, this is not a big issue, also considering that you will change the structure of folders and so.

3) I started the the TEST1 script in producers/calice/runscripts (changing the euRun.exe by euRun and same for the data collector) and it connects ok to everything (logger, run control, data collector and producer). The problem comes when I configure: I do it with the usual configuration file that I use ( producers/calice/conf/testSpiroc2d.conf) and then it gives me the exception:

Configuring (testSpiroc2d)...
Command Thread exiting due to exception:
trying to push Empty Processor to batch
  From /home/calice/eudaq2-arummler/main/lib/processors/Processor_batch.cc:81
  In void eudaq::Processor_batch::pushProcessor(eudaq::Processor_rp)

Anyhow, it sends the start/stop command to our system but the eudaq does not process the data and does not try to save the data in any format (native, calice, etc).

Maybe I should do some change in the producer in order to adapt to eudaq2... Are there other producers tested in eudaq 1 and 2? Did they need to implement any new feature?

4) Why the runnumber.dat file is missing? Does this mean that during testbeam, we wont be able to keep track of the runnumbers if the eudaq is closed/crashed? Or is this implemented in other way?


Cheers,
Adrian

**Comment by simonspa** [comment]
I'm merging this since the issues mentioned are neither introduced nor addressed by this PR.
@airqui Please collect the single items as issues so we can address them one-by-one. Thanks!

**Comment by arummler** [comment]
Opened issue #181 for 2.) Thank you for observing this. Will commit a fix soon.

**Pull request merged**